### PR TITLE
[pyspark][doc] add more doc for pyspark

### PR DESCRIPTION
**Pull request by wbo4958**
Add doc for pyspark gpu support.

### CODE REVIEW

**Comment by wbo4958** [comment]
@WeichenXu123 @trivialfis please help to review this PR. Thx

File: doc/tutorials/spark_estimator.rst [hunk 1]
```diff
@@ -51,17 +63,141 @@ type or spark array type.
...
+  # predict on test data
+  predict_df = model.transform(test_df)
+  predict_df.show()
+
+Prepare the necessary packages
+==============================
+
+We recommend using Conda or Virtualenv to manage python dependencies
+in PySpark. Please refer to
+`How to Manage Python Dependencies in PySpark <https://www.databricks.com/blog/2020/12/22/how-to-manage-python-dependencies-in-pyspark.html>`_.
```
Comment (trivialfis) [inline doc/tutorials/spark_estimator.rst:130]: This varies between different providers of pyspark environments. On dataproc we can't submit the environment through spark-submit.

File: doc/tutorials/spark_estimator.rst [hunk 2]
```diff
@@ -51,17 +63,141 @@ type or spark array type.
...
+
+.. code-block:: python
+
+  # save the model
+  model.save("/tmp/xgboost-pyspark-model")
+
+  # load the model
+  model2 = SparkXGBRankerModel.load("/tmp/xgboost-pyspark-model")
+
+The above code snippet shows how to save/load xgboost pyspark model. And you can also
```
Comment (trivialfis) [inline doc/tutorials/spark_estimator.rst:171]: I strongly prefer using the booster attribute and would like to keep this special file name as a workaround that should be used sparingly.

File: doc/tutorials/spark_estimator.rst [hunk 3]
```diff
@@ -51,17 +63,141 @@ type or spark array type.
...
+
+
+Accelerate the whole pipeline of xgboost pyspark
+================================================
+
+With `RAPIDS Accelerator for Apache Spark <https://nvidia.github.io/spark-rapids/>`_,
+you can accelerate the whole pipeline (ETL, Train, Transform) for xgboost pyspark
+without any code change by leveraging GPU.
+
+You only need to add some configurations to enable RAPIDS plugin when submitting.
```
Comment (trivialfis) [inline doc/tutorials/spark_estimator.rst:188]:
```suggestion
You only need to add some configurations to enable RAPIDS plugin when submitting.
```
Below is a simple example submit command for enabling GPU acceleration:

**Review by WeichenXu123 — APPROVED** [review]
LGTM

File: doc/tutorials/spark_estimator.rst (hunk 1 above)
Comment (wbo4958) [reply doc/tutorials/spark_estimator.rst:130]: I mentioned this tutorial is for spark standalone mode, I didn't want to involve other CSPs to xgboost.

File: doc/tutorials/spark_estimator.rst (hunk 2 above)
Comment (wbo4958) [reply doc/tutorials/spark_estimator.rst:171]: Hmm, this is the standard spark way to save/load model. I can also mention the booster attribute.

**Comment by wbo4958** [comment]
@trivialfis please help to review it again. Thx

**Review by trivialfis — APPROVED** [review]

**Comment by trivialfis** [comment]
@wbo4958 I made some modifications to your PR, do you have any concerns?

**Comment by wbo4958** [comment]
It looks pretty much better than my original expression. Thx

**Pull request merged**